From 389276ef566342c3c61547326c070a0d6914622a Mon Sep 17 00:00:00 2001
From: Jason Gerecke <killertofu@gmail.com>
Date: Thu, 23 Feb 2012 19:05:13 +0000
Subject: wacom: Make touch devices insensitve to KEY_IS_ABSOLUTE changes

Pen devices are typically used in absolute mode, while touch
devices are used in relative mode. However, the tracking mode
of *both* currently depend on the value of KEY_IS_ABSOLUTE.
If a preference is set for absolute or relative it will be
applied to both devices.

This patch has touch devices ignore changes to KEY_IS_ABSOLUTE.
Ideally it would be nice to have this key available independently
for both pen and touch devices at some point in the future.

https://bugzilla.gnome.org/show_bug.cgi?id=670655
---
diff --git a/plugins/wacom/gsd-wacom-manager.c b/plugins/wacom/gsd-wacom-manager.c
index 05a83b9..250ea84 100644
--- a/plugins/wacom/gsd-wacom-manager.c
+++ b/plugins/wacom/gsd-wacom-manager.c
@@ -557,7 +557,8 @@ wacom_settings_changed (GSettings      *settings,
 		set_tpcbutton (device, g_settings_get_boolean (settings, key));
 	} else if (g_str_equal (key, KEY_IS_ABSOLUTE)) {
 		if (type != WACOM_TYPE_CURSOR &&
-		    type != WACOM_TYPE_PAD)
+		    type != WACOM_TYPE_PAD &&
+		    type != WACOM_TYPE_TOUCH)
 			set_absolute (device, g_settings_get_boolean (settings, key));
 	} else if (g_str_equal (key, KEY_AREA)) {
 		if (type != WACOM_TYPE_CURSOR &&
--
cgit v0.9.0.2

